components/brave_wallet_ui/page/screens/swap/components/swap/swap_providers/swap_providers.tsx

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+import * as React from 'react'
+
+// Selectors
+import {
+  useSafeUISelector //
+} from '../../../../../../common/hooks/use-safe-selector'
+import { UISelectors } from '../../../../../../common/selectors'
+
+// Constants
+import LPMetadata from '../../../constants/LpMetadata'
+
+// Utils
+import { getLocale } from '../../../../../../../common/locale'
+
+// Types
+import {
+  BraveWallet,
+  SupportedSwapProviders,
+  SwapProviderNameMapping
+} from '../../../../../../constants/types'
+
+// Components
+import {
+  InfoIconTooltip //
+} from '../../../../../../components/shared/info_icon_tooltip/info_icon_tooltip'
+
+// Styles
+import { RadioButton, ProviderIcon } from './swap_providers.style'
+import {
+  Column,
+  HorizontalSpace,
+  LeoSquaredButton,
+  Row,
+  Text
+} from '../../../../../../components/shared/style'
+
+interface Props {
+  selectedProvider: BraveWallet.SwapProvider
+  onChangeSwapProvider: (provider: BraveWallet.SwapProvider) => void
+  availableProvidersForSwap: BraveWallet.SwapProvider[]
+}
+
+export const SwapProviders = (props: Props) => {
+  const { selectedProvider, onChangeSwapProvider, availableProvidersForSwap } =
+    props
+
+  // Selectors
+  const isPanel = useSafeUISelector(UISelectors.isPanel)
+
+  // State
+  const [userSelectedSwapProvider, setUserSelectedSwapProvider] =
+    React.useState<BraveWallet.SwapProvider>(selectedProvider)
+
+  // Memos
+  const providerList = React.useMemo(() => {
+    return SupportedSwapProviders.sort(
```

Comment:
Let's not mutate the original array.

```suggestion
    return [...SupportedSwapProviders].sort(
```

Reply:
Done!